### DISCUSSION TITLE
Set cookies is not working while using RedirectResponse

### DISCUSSION THREAD

**Discussion by pythonhubdev** [fastapi/fastapi#6320] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/route")
async def internalLogin(request: Request, response: Response,
                        data: Model = Depends(Model.asForm)):
    try:
        res, token = await openApiController(data=data)
        if res:
            response.set_cookie(key="token", value=token)
            return RedirectResponse(url="/route", status_code=status.HTTP_303_SEE_OTHER)
        else:
            return templates.TemplateResponse("401.html", {"request": request})
    except DocumentNotFound as documentNotFoundException:
        print(documentNotFoundException)
```

### Description

I am trying to set cookie after the response is over. But when I call `set_cookie` method I am not able to see the cookies in the developer console also I am not even to print the cookie that is being stored. But if I am not using `RedirectResponse` the set cookie method works perfectly but it doesn't work when I use `RedirectResponse`.

### Operating System

macOS

### FastAPI Version

0.73

### Python Version

3.10

**Comment by rogerbrinkmann** [comment, marked as answer]
@pythonhubpy, you are returning a newly created response object which doesn't have the cookie set.
Try the following: (untested)
```python
if res:
    response = RedirectResponse(url="/route", status_code=status.HTTP_303_SEE_OTHER)
    response.set_cookie(key="token", value=token)
    return response
```

**Reply by cesarqdt** [reply]
this does not work for me:

```
access_token = result.get("access_token")
query_params = urlencode({"email": user.email})
redirect_url = f"{LOGIN_ACCESS_TOKEN_PATH}?{query_params}"

response = RedirectResponse(url = redirect_url, status_code = status.HTTP_303_SEE_OTHER)
response.set_cookie(key = "access_token", value = access_token)

return response
```

not working.

**Comment by pythonhubdev** [comment]
This worked thanks.
